### PR TITLE
Discard the space after colon in HTTP respone headers

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**
For example, for a response header `Content-Encoding: gzip`, intuitively people would expect the value without the leading space when trying to get the value of content encoding.

### CODE REVIEW

File: sdk/platform/http_client/curl/src/curl_client.cpp [hunk 1]
```diff
@@ -99,6 +99,10 @@ static void ParseHeader(std::string const& header, std::unique_ptr<Response>& re
 
   auto headerName = std::string(start, end);
   start = end + 1; // start value
+  if (start < header.end() && *start == ' ')
```
Comment (ahsonkhan) [inline sdk/platform/http_client/curl/src/curl_client.cpp:102]: What about multiple whitespace? An what about other whitespace characters like horizontal tab? We should see what the HTTP RFC states about this:
https://tools.ietf.org/html/rfc7230#section-3.2

> header-field   = field-name ":" OWS field-value OWS

> OWS            = *( SP / HTAB )
                    ; optional whitespace

Comment (JinmingHu-MSFT) [reply sdk/platform/http_client/curl/src/curl_client.cpp]: @ahsonkhan updated.

File: sdk/platform/http_client/curl/src/curl_client.cpp [hunk 2]
```diff
@@ -99,6 +99,10 @@ static void ParseHeader(std::string const& header, std::unique_ptr<Response>& re
 
   auto headerName = std::string(start, end);
   start = end + 1; // start value
+  while (start < header.end() - 2 && (*start == ' ' || *start == '\t'))
```
Comment (ahsonkhan) [inline sdk/platform/http_client/curl/src/curl_client.cpp:102]: What does the `-2` here mean? Maybe a comment would be helpful.

Also, @vhvb1989 - we should certainly have tests for these :)

Comment (JinmingHu-MSFT) [reply sdk/platform/http_client/curl/src/curl_client.cpp]: @ahsonkhan I removed the ` -2` so it's more intuitive.

**Comment by JinmingHu-MSFT** [comment]
Can anybody merge this PR since we're blocking here?

**Pull request merged**